### ISSUE TITLE
Conan graph build order accepting a list of lockfiles to consolidate build order

### ISSUE THREAD

**Issue by czoido** [conan-io/conan#7166]
Hi,

I think that for certain cases in CI flows it would be useful to be able to get a build order across  multiple profiles for the same library. Now, the build order works using a single lockfile (corresponding to one profile) as an input:

`conan graph build-order lockfile-release.lock --build missing --json=build_order.json`

In the CI we are going to have multiple configurations that we would like to build. So, for example,  if you want to use the build order to trigger pipelines that build the libraries in the CI, you have to trigger that pipeline several times, one for each configuration. Imagine a graph like this:

![image](https://user-images.githubusercontent.com/5045666/84114621-fa217100-aa2c-11ea-9250-fe2b6fba4fdb.png)

```
conan graph build-order lockfile-release.lock --build missing ...
-> the build order says you have to build libz 
-> trigger the pipeline for libz using lockfile-release-linux.lock
-> continue building the rest of the graph with the build order

conan graph build-order lockfile-debug.lock --build missing ...
-> the build order says you have to build libz 
-> trigger the pipeline for libz using lockfile-debug-linux.lock
-> continue building the rest of the graph with the build order
```

Normally, in my library's pipeline I have a list of configurations I want to build, so it would be nice that there was a way that `conan graph build-order` accepted a list of coherent lockfiles and calculated a consolidated build order for all of them:

`conan graph build-order lockfile-release.lock lockfile-debug.lock --build missing`

We could get something like:

```
[
    [("libz/1.0#9cf70cfbdc4dfd829e3cbc415a66ad3e",
      ["lockfile-release.lock",
       "lockfile-debug.lock"
      ])
    ],
    [("liba/1.0#829e3cbc415a69cf70cfbdc4dfd6ad3e",
      ["lockfile-release.lock",
       "lockfile-debug.lock"
      ])
    ],
    [[("libb/1.0#b0782f7ab39824c2148e722d659a9bc4",
      ["lockfile-release.lock",
       "lockfile-debug.lock"
      ]),
    ("libc/1.0#e722d6b0ab39824c2148782f759a9bc4",
      ["lockfile-release.lock",
       "lockfile-debug.lock"
      ])
    ]],
….
]
```

The in the CI it would be possible to build all the graph for different configurations in one pass.

```
conan graph build-order lockfile-release.lock lockfile-debug.lock  --build missing ...
-> the build order says you have to build libz for release and debug
-> trigger the pipeline for libz using lockfile-release-linux.lock and lockfile-debug.lock
-> continue building the rest of the graph with the build order for multiple configs
```

This way you don't have to trigger again the same pipeline just for building a new configuration.

There will be cases like having conditional requirements that should be taken into account (imagine an `option: with_libz`)

![image](https://user-images.githubusercontent.com/5045666/84115540-85e7cd00-aa2e-11ea-9e16-d819c3cec53c.png)

The you would get first the `libz` for your profiles with `libz` enabled and then liba for the rest and so on...

**Comment by solvingj** [comment]
The timing on this could not be more perfect.  I stopped work on a CI flow last night after reaching the point where I had gathered all the lockfiles for all configurations of a pipeline, and then generated a list of `build-order.json` files.  My next step today was going to be to iterate over all those `build-order.json` files and calculate the "collective build order".  This logic was going to be non-trivial, and would be MUCH more desirable if this was internalized into a feature.  In principle, your example looks perfect.
